doc/modules/cross_validation.rst

```diff
@@ -467,44 +378,181 @@ Here is a usage example::
 validation that allows a finer control on the number of iterations and
 the proportion of samples on each side of the train / test split.
 
+Cross-validation iterators with stratification based on class labels.
+=====================================================================
+
+In some datasets the number of samples per class varies based on the class
+label. They are referred to commonly as datasets with imbalance in classes.
+In such cases, stratified sampling may be done to pick the samples for
+cross-validation. This avoids the class imbalance from affecting the
+cross-validation results.
 
-Label-Shuffle-Split
+The :class:`StratifiedKFold` and :class:`StratifiedShuffleSplit` do
+stratified cross-validation to ensure that the proportion of samples from each
+class is same as far as possible.
+
+Stratified k-fold
+-----------------
+
+:class:`StratifiedKFold` is a variation of *k-fold* which returns *stratified*
+folds: each set contains approximately the same percentage of samples of each
+target class as the complete set.
+
+Example of stratified 3-fold cross-validation on a dataset with 10 samples from
+two slightly unbalanced classes::
+
+  >>> from sklearn.model_selection import StratifiedKFold
+
+  >>> X = np.ones(10)
+  >>> y = [0, 0, 0, 0, 1, 1, 1, 1, 1, 1]
+  >>> skf = StratifiedKFold(n_splits=3)
+  >>> for train, test in skf.split(X, y):
+  ...     print("%s %s" % (train, test))
+  [2 3 6 7 8 9] [0 1 4 5]
+  [0 1 3 4 5 8 9] [2 6 7]
+  [0 1 2 4 5 6 7] [3 8 9]
+
+Stratified Shuffle Split
+------------------------
+
+:class:`StratifiedShuffleSplit` is a variation of *ShuffleSplit*, which returns
+stratified splits, *i.e* which creates splits by preserving the same
+percentage for each target class as in the complete set.
+
+.. _group_cv
+
+Cross-validation iterators for grouped data.
+============================================
+
+Data is non-i.i.d. when all the samples are not drawn from an identical
+distribution. In some cases such data can be grouped into individual groups.
+The samples belonging to a particular group are identically distributed within
+their group.
+
+The grouping is domain specific. An example would be when there is
+medical data collected from a few patients, with multiple samples taken from
+each patient. Such data is likely to be dependent on the individual patient.
+In other words, all the samples for a particular patient are from the same
+identical distribution. Hence, in this case the patient id will be the group
+identifiers.
```

Comment:
Yeah I think it's more the core of what this tries to do. Often you want to generalize to new groups. Then you need this. If you are not interested in new groups, doing this will be overly pessimistic.